types/prosemirror-model/index.d.ts

```diff
@@ -671,6 +671,10 @@ declare class ProsemirrorNode<S extends Schema = any> {
      * uneditable [node view](#view.NodeView)).
      */
     isAtom: boolean;
+    /**
+     * The node type's [whitespace](#view.NodeSpec.whitespace) option.
+     */
+    whitespace: 'pre' | 'normal';
```

Comment:
This is a [getter](https://github.com/ProseMirror/prosemirror-model/commit/b9085fbf195e07f336c48647a4b318c8d6133054#diff-76078cd7448ebeb6bb584d32f1ca27b7d2001d779ab0b68f956fbcb8791a473c), so I'd make this property `readonly`. Other than that it looks good to me.

Reply:
None of the other existing properties like `isInline`, `isTextblock`, `isLeaf`, `isAtom` is marked as `readonly`. That's why I didn't do that either. That should probably be done in a separate PR for all properties then.